test-result-summary-client/src/PerfCompare/PerfCompare.jsx

```diff
@@ -206,6 +226,13 @@ export default class PerfCompare extends Component {
                 resBenchmarkTestJson.testInfo === undefined) {
             displayErrorMessage += "Test build not found"
         }
+        // Check if the aggregate info is valid
+        if (!resBenchmarkBaselineJson.testInfo || !resBenchmarkBaselineJson.testInfo[0].aggregateInfo || resBenchmarkBaselineJson.testInfo[0].aggregateInfo.length < 1 || resBenchmarkBaselineJson.testInfo[0].aggregateInfo[0].metrics.length === 0) {
+            displayErrorMessage += "Baseline build has no data. "
+        }
+        if (!resBenchmarkTestJson.testInfo || !resBenchmarkTestJson.testInfo[0].aggregateInfo || resBenchmarkTestJson.testInfo[0].aggregateInfo.length < 1 || resBenchmarkTestJson.testInfo[0].aggregateInfo[0].metrics.length === 0) {
+            displayErrorMessage += "Test build has no data. "
+        }
```

Comment:
The code is very verbose. Could we do try catch here? All we want to do is to verify resBenchmarkTestJson.testInfo[0].aggregateInfo[0].metrics.length > 0

Reply:
Updated!